packages/client/src/generation/TSClient/PrismaClient.ts

```diff
@@ -464,7 +464,7 @@ export class PrismaClientClass implements Generable {
     return `${this.jsDoc}
 export class PrismaClient<
   ClientOptions extends Prisma.PrismaClientOptions = Prisma.PrismaClientOptions,
-  U = 'log' extends keyof ClientOptions ? ClientOptions['log'] extends Array<Prisma.LogLevel | Prisma.LogDefinition> ? Prisma.GetEvents<ClientOptions['log']> : never : never,
+  const U = 'log' extends keyof ClientOptions ? ClientOptions['log'] extends Array<Prisma.LogLevel | Prisma.LogDefinition> ? Prisma.GetEvents<ClientOptions['log']> : never : never,
```

Comment:
This is a breaking change. Const generics require TypeScript 5.0 or later but Prisma supports 4.7+ right now. We're raising the minimal required version to 5.1 soon in Prisma 6 though.

Reply:
Yeah... Unfortunately it (`Const generics require TypeScript 5.0 or later`) does...
But you can cherry pick one commit and not touch the other one.
Or I can create 2 split PRs. One of them for the closest version and the second one is for 6.0
What do you think?

Reply:
Prisma 6 is around the corner so let's just merge the whole PR.